Examples/HelloWorld/HelloWorld.cpp

```diff
@@ -27,23 +27,94 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 // HelloWorld.cpp - the simplest possible demo example of CacheSim
 
 #include "CacheSim/CacheSim.h"
+#include <cmath>
+#include <string>
+#include <list>
+#include <thread>
+
+CacheSim::DynamicLoader cachesim;
+volatile bool run = true;
+volatile bool workFinished = false;
+volatile bool canExit = false;
+
+void DoSomeWork()
```

Comment:
I feel this example is complicated enough that it doesn't belong in "Hello world" anymore. Perhaps split it out into a ThreadedExample project?